junit-jupiter-params/src/main/java/org/junit/jupiter/params/aggregator/ArgumentsAccessor.java

```diff
@@ -45,6 +45,14 @@
 @API(status = STABLE, since = "5.7")
 public interface ArgumentsAccessor {
 
+	/**
+	 * Get the zero-based invocation index of the ParameterizedTest arguments source.
+	 *
+	 * @return the invocation index of the arguments source.
+	 * @since 5.9
+	 */
+	long getInvocationIndex();
```

Comment:
```suggestion
	@API(status = EXPERIMENTAL, since = "5.9")
	long getInvocationIndex();
```

plus import

Reply:
Why is it zero-based?

Reply:
I think an `int` would suffice.

Reply:
This value is derived from `AtomicLong invocationCount` in `provideTestTemplateInvocationContexts` of `ParameterizedTestExtension`. I didn't want to mix types. I could change it to AtomicInteger if that is more realistic.

Reply:
I'll change it to start it at 1.

Reply:
Please let me know if you'd like the `invocationCount` converted to `AtomicInteger`.